service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/ClientInstallationServiceImpl.java

```diff
@@ -168,6 +168,10 @@ public Installation findById(String primaryKey) {
 
     @Override
     public void removeInstallation(Installation installation) {
+        //Let manually remove the categories and update the installations before deleting
+        //check AGPUSH-1295 for more information
+        installation.setCategories(null);
+        updateInstallation(installation);
```

Comment:
Instead of manually updating the categories before deleting couldn't we change the cascade setting?
https://github.com/aerogear/aerogear-unifiedpush-server/blob/159c119c6ef1564846b847e6b418335c2ce01dc0/model/jpa/src/main/resources/META-INF/orm.xml#L51
to
```xml
<cascade>
   <cascade-persist/>
</cascade>
```